File: compare_version_numbers.py
class Solution:
    # @param version1, a string
    # @param version2, a string
    # @return an integer
    def compareVersion(self, version1, version2):
        v1 = [int(k) for k in version1.split('.')]
        v2 = [int(k) for k in version2.split('.')]
        if len(v1) < len(v2):
            v1 += [0] * ( len(v2) - len(v1) ) 
        if len(v1) > len(v2):
            v2 += [0] * ( len(v1) - len(v2) ) 
        for i in xrange(len(v1)):
            if v1[i] > v2[i]:
                return 1
            elif v1[i] < v2[i]:
                return -1
        return 0

File: Leetcode_Java/CompareVersionNumbers.java
import java.util.*;

public class CompareVersionNumbers {
    public int compareVersion(String version1, String version2) {
        String[] versionOne = version1.split("\\.");
        // take care!!! "." can match any character in regular expression!!!
        String[] versionTwo = version2.split("\\.");
        if (versionOne.length > versionTwo.length) {
            return -1 * compareVersion(version2, version1);
            // version1 and version2 is swapped, so we need to multiply the result by -1
        }
        String[] versionOneExtend = new String[versionTwo.length];
        for (int i = 0; i < versionTwo.length; i++) {
            if (i < versionOne.length) {
                versionOneExtend[i] = versionOne[i];
            } else {
                versionOneExtend[i] = "0";
            }
        }
        int i = 0;
        while (i < versionTwo.length) {
            int one = Integer.parseInt(versionOneExtend[i]);
            int two = Integer.parseInt(versionTwo[i]);
            if (one > two) {
                return 1;
            } else if (one < two) {
                return -1;
            } else {
                i++;
            }
        }
        return 0;
    }

    public static void main(String[] args) {
        CompareVersionNumbers test = new CompareVersionNumbers();
        System.out.println(test.compareVersion("1.4.3.0","1.4.3"));
        //System.out.println(Arrays.toString("1.4.3".split("\\.")));
    }
}